# Generated by Django 4.0.4 on 2022-05-07 00:39

from django.db import migrations, models


class Migration(migrations.Migration):

    dependencies = [
        ('board', '0003_alter_task_error'),
    ]

    operations = [
        migrations.CreateModel(
            name='Wall',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('type', models.CharField(max_length=100)),
                ('x', models.IntegerField()),
                ('y', models.IntegerField()),
                ('closed', models.BooleanField(default=False)),
            ],
        ),
        migrations.AlterModelOptions(
            name='player',
            options={'verbose_name': 'Player', 'verbose_name_plural': 'Players'},
        ),
        migrations.AlterField(
            model_name='task',
            name='error',
            field=models.CharField(blank=True, max_length=256, null=True),
        ),
        migrations.AddConstraint(
            model_name='wall',
            constraint=models.UniqueConstraint(fields=('type', 'x', 'y'), name='unique position and direction'),
        ),
    ]
